File: iPhone/SoundRecorder/Classes/Core/SoundEditor/Internal/SERecord.m
//  SERecord.m
//  SoundRecorder//  Created by Igor Danich igor.danich@wise-apps.com on 11.04.14.

#import "SERecord.h"

#import "SERecordAudioStream.h"

@interface SERecord()
@property(nonatomic,strong) SERecordAudioStream* pv_stream;
@end

@implementation SERecord

- (SERecordAudioStream*)audioStream {
    if (!self.pv_stream) {
        self.pv_stream = [[SERecordAudioStream alloc] initWithRecord:self];
    }
    return self.pv_stream;
}

- (void)updateToDictionary:(NSDictionary*)dictionary {
    self.soundURL = [NSURL URLWithString:dictionary[@"soundPath"]];
    SERecordSoundRange range;
    range.start = [dictionary[@"start"] integerValue];
    range.duration = [dictionary[@"duration"] integerValue];
    self.soundRange = range;
}

- (NSMutableDictionary*)dictionaryRepresentation {
    NSMutableDictionary* dict = [super dictionaryRepresentation];
    if (self.soundURL) {
        dict[@"soundPath"] = [self.soundURL absoluteString];
    }
    dict[@"start"] = @(self.soundRange.start);
    dict[@"duration"] = @(self.soundRange.duration);
    return dict;
}

@end

File: Blackberry/SoundRecorder/src/core/se/SERecord.cpp
/*
 * SERecord.cpp
 *
 *  Created on: 16.04.2014
 *      Author: Timofey Kovalenko (timothy.kovalenko@wise-apps.com)
 */

#include "SERecord.h"

SERecord::SERecord():soundUrl("") {
	soundRange.start = 0;
	soundRange.duration = 0;
    m_audio_stream = NULL;


}

SERecord::~SERecord() {
    free(m_audio_stream);
}

SERecordAudioStream* SERecord::audioStream() {

//	SERecordAudioStream stream;
//	stream.initWithRecord(this);
//	m_audio_stream = stream;
    if(m_audio_stream == NULL) {
        m_audio_stream = new SERecordAudioStream();
        //m_audio_stream->initWithRecord(this);
    }
    m_audio_stream->initWithRecord(this);
    return m_audio_stream;
}
